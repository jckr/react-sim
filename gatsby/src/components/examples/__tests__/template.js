import React from 'react';
import renderer from 'react-test-renderer';
import { initData, updateData, draw, Frame, params } from '../template';
import ctx from '../../helpers/mock-ctx';
import Random from '../../helpers/random';

describe('template', () => {
  let data;
  it('data function works', () => {
    const rng = new Random();
    data = initData(params, rng.get);
    expect(data).toMatchSnapshot();
    const complete = jest.fn();
    for (let tick = 1; tick < 5; tick++) {
      data = updateData({ data, tick, params, complete }, rng.get);
      expect(data).toMatchSnapshot();
    }
  }),
    it('model renders properly', () => {
      const frame = renderer
        .create(
          <Frame data={data} tick={5} ctx={ctx} params={params} />
        )
        .toJSON();
      expect(frame).toMatchSnapshot();
    });
});
