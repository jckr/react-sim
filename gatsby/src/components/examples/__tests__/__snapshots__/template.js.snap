// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template data function works 1`] = `undefined`;

exports[`template data function works 2`] = `undefined`;

exports[`template data function works 3`] = `undefined`;

exports[`template data function works 4`] = `undefined`;

exports[`template data function works 5`] = `undefined`;

exports[`template model renders properly 1`] = `<canvas />`;
